tests/test_client.py

```diff
@@ -0,0 +1,197 @@
+                                                   environment={"MINIO_ROOT_USER": "minioadmin",
+                                                                "MINIO_ROOT_PASSWORD": "minioadmin",
+                                                                "MINIO_UPDATE": "off"},
+                                                   ports={"9000": None}, detach=True)
+
+        cls.docker1.reload()
+        cls.docker2.reload()
+        docker1_port = cls.docker1.ports['9000/tcp'][0]['HostPort']
+        docker2_port = cls.docker2.ports['9000/tcp'][0]['HostPort']
+
+        cls.minio1 = boto3.client('s3',
+                                  endpoint_url=f'http://localhost:{docker1_port}',
+                                  aws_access_key_id='minioadmin',
+                                  aws_secret_access_key='minioadmin', )
+
+        cls.minio2 = boto3.client('s3',
+                                  endpoint_url=f'http://localhost:{docker2_port}',
+                                  aws_access_key_id='minioadmin',
+                                  aws_secret_access_key='minioadmin')
+
+        cls.botwo_client = create_botwo_client(cls.minio1, cls.minio2)
+
+    def test_upload(self):
+        self.minio1.create_bucket(Bucket="test-upload")
+        content = "TEST_UPLOAD"
+        self.botwo_client.put_object(Bucket="bucket-a", Key="a/b/1.txt", Body=content)  # route to minio1
+
+        data = self.botwo_client.get_object(Bucket="bucket-a", Key="a/b/1.txt")
+        contents = data['Body'].read().decode('utf-8')
+        self.assertEqual(contents, content)
+
+        with pytest.raises(Exception):
+            # assert that the call is to minio1
+            self.minio2.get_object(Bucket="test-upload", Key="main/a/b/1.txt")
+
+    def test_copy(self):
+        self.minio1.create_bucket(Bucket="test-copy")
+        content = "TEST_COPY"
+        self.botwo_client.put_object(Bucket="test-copy", Key="main/a/b/1.txt", Body=content)  # route to minio1
+        self.botwo_client.copy_object(Bucket="test-copy", Key="1.txt",
+                                      CopySource={"Bucket": "source", "Key": "a/b/1.txt"})
+
+        data = self.botwo_client.get_object(Bucket="test-copy", Key="1.txt")
+        contents = data['Body'].read().decode('utf-8')
+        self.assertEqual(contents, content)
+
+        with pytest.raises(Exception):
+            # assert that the call is to minio1
+            self.minio2.get_object(Bucket="test-copy", Key="1.txt")
+
+    def test_default(self):
+        self.minio2.create_bucket(Bucket="test-default")
+        content = "TEST_DEFAULT"
+        self.botwo_client.put_object(Bucket="test-default", Key="2.txt", Body=content)  # route to default minio2
+
+        data = self.botwo_client.get_object(Bucket="test-default", Key="2.txt")
+        contents = data['Body'].read().decode('utf-8')
+        self.assertEqual(contents, content)
+
+        with pytest.raises(Exception):
```

Comment:
This is Python!  Need to test a particular exception, otherwise even some syntax and logic errors in the called code (or in the test code) will make the test pass.

Also in every call to `pytest.raises(Exception)`